axelrod/tests/strategies/test_meta.py

```diff
@@ -148,6 +152,42 @@ def test_strategy(self):
         self.assertEqual(P1.strategy(P2), C)
 
 
+class TestMetaWinnerEnsemble(TestMetaPlayer):
+    name = "Meta Winner Ensemble"
+    player = axl.MetaWinnerEnsemble
+
+    def test_singularity(self):
+        """Test meta_strategy when the player is singular."""
+        team = [axl.Cooperator]
+        player = axl.MetaWinnerEnsemble(team=team)
+        self.assertFalse(Classifiers["stochastic"](player))
+        coplayer = axl.Defector()
+        match = axl.Match((player, coplayer), turns=10)
+        match.play()
+
+    def test_stochasticity(self):
+        # One player teams may be stochastic or not
+        team = [axl.Cooperator]
+        player = axl.MetaWinnerEnsemble(team=team)
+        self.assertFalse(Classifiers["stochastic"](player))
+
+        team = [axl.Random]
+        player = axl.MetaWinnerEnsemble(team=team)
+        self.assertTrue(Classifiers["stochastic"](player))
+
+        # Multiplayer teams without repetition are always stochastic
+        team = [axl.Cooperator, axl.Defector]
+        player = axl.MetaWinnerEnsemble(team=team)
+        self.assertTrue(Classifiers["stochastic"](player))
+
+        # If the players are all identical, a multiplayer
+        # team might is in fact deterministic, even though random values
+        # are being drawn.
```

Comment:
On a more general note: is this wanted behaviour? Not to be done in this PR, but we could add a check to the meta player for this?

Reply:
It's a weird edge case for sure but it makes sense I guess.

Reply:
Yeah I see that :+1:

Reply:
I also made changes to the strategy code to not actually draw any random numbers in this case. `Match` will now only set a seed if a `Player` needs it, determined by whether the strategy is classified as stochastic or not. There are many changes to get the classifiers right, and if it's wrong, there should be a warning or exception thrown as a strategy attempts to use `self._random` when it doesn't exist.

Reply:
Nice.